pkg/storegateway/indexcache/cache.go

```diff
@@ -116,6 +120,23 @@ type IndexCache interface {
 	FetchLabelValues(ctx context.Context, userID string, blockID ulid.ULID, labelName string, matchersKey LabelMatchersKey) ([]byte, bool)
 }
 
+// BlockTTL determines an appropriate TTL for cache entries related to a particular block
+// based on metadata about the block. Use a shorter TTL for temporary blocks that will be
+// compacted and deleted soon and a longer TTL for larger blocks that won't be deleted.
+func BlockTTL(meta *block.Meta) time.Duration {
+	duration := time.Duration(meta.MaxTime-meta.MinTime) * time.Millisecond
+	duration = max(time.Hour, duration.Round(time.Hour))
```

Comment:
[sounds a nit but may be critical if we'll reduce blocks duration] I would ceil instead of round.

Reply:
Done!